cjworkbench/i18n/views.py

```diff
@@ -20,10 +21,11 @@ def set_locale(request):
         url=next, allowed_hosts={request.get_host()}, require_https=request.is_secure()
     ):
         next = "/"
+    response = HttpResponseRedirect(next)
     locale = request.POST.get("new_locale")
     if is_supported(locale):
-        request.session["locale_id"] = locale
+        response.set_cookie(COOKIE_NAME, locale)
```

Comment:
Add `max_age=365*86400`. Otherwise the cookie disappears when the user closes the browser window.